torchdata/nodes/samplers/utils.py

```diff
@@ -10,11 +10,15 @@
 import torch.distributed as dist
 
 
-def _get_rank_seed(seed: int, generator_rank: torch.Generator, rank: int, world_size: int) -> int:
-    generator_rank.manual_seed(seed * world_size + rank)
+def _get_rank_seed(seed: int, generator_rank: torch.Generator, rank: int, world_size: int, epoch: int) -> int:
+    generator_rank.manual_seed(seed * world_size + rank + epoch)
```

Comment:
Will this cause create a pattern between ranks when going across epochs?

eg)
                                Epoch ->
Seeds for Rank 0/2: [b, c, d, e, f, g, a, ...]
Seeds for Rank 1/2: [a, b, c, d, e, f, g, h, ...]

Reply:
Maybe something like this instead might work:
```suggestion
    generator_rank.manual_seed(seed * world_size + rank)
    (assuming epoch is incremented instead of randomly generated)
    for _ in range(epoch):
      __consumes_rng_state(generator_rank)
```